lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php

```diff
@@ -1025,7 +1024,11 @@ public function validateIdentifier()
     public function validateAssociations()
     {
         foreach ($this->associationMappings as $mapping) {
-            if ( ! ClassLoader::classExists($mapping['targetEntity']) ) {
+            if (
+                ! class_exists($mapping['targetEntity'])
+                && ! interface_exists($mapping['targetEntity'])
+                && ! trait_exists($mapping['targetEntity'])
```

Comment:
I don't think targeting a trait is actually supported btw

Reply:
Agreed, kept it for compat with `ClassLoader::classExists()`.